cmd/dosa/schema.go

```diff
@@ -62,6 +62,15 @@ func (c *SchemaCmd) doSchemaOp(name string, f func(dosa.AdminClient, context.Con
 		fmt.Printf("options are %+v\n", *c)
 		fmt.Printf("global options are %+v\n", options)
 	}
+
+	// if not given, set the service name dynamically based on scope
+	if options.ServiceName == "" {
```

Comment:
nit: this is really a mapping of scope->service that should be easier to edit than this; recommend an array of globs to match against, with "*" as the final one:

```
type ScopeServiceMap struct {
    globPattern string
    serviceName string
} []scopeServiceMap = {
 { "production", "dosa-gateway" },
 { "*", "dosa-dev-gateway" },
}
```